### PR TITLE
Static fused moe op

### PR DESCRIPTION
**Pull request by jkaniecki**
Added static fused_moe OP to enable mixtral model on Gaudi.

### CODE REVIEW

File: vllm/model_executor/models/mixtral.py [hunk 1]
```diff
@@ -220,28 +220,36 @@ def process_weights_after_loading(self):
                                          requires_grad=False)
 
     def forward(self, hidden_states: torch.Tensor) -> torch.Tensor:
-        num_tokens, hidden_size = hidden_states.shape
+        batch_size, sequence_length, hidden_size = hidden_states.shape
```
Comment (kzawora-intel) [inline vllm/model_executor/models/mixtral.py:223]: this is HPU specific and will break on other accelerators. Please cover that with is_hpu.

File: vllm/model_executor/models/mixtral.py [hunk 2]
```diff
@@ -220,28 +220,36 @@ def process_weights_after_loading(self):
...
+                                            w2_scale=self.w2_scale,
+                                            a1_scale=self.a13_scale,
+                                            a2_scale=self.a2_scale)
 
         if self.tp_size > 1:
             final_hidden_states = tensor_model_parallel_all_reduce(
                 final_hidden_states)
 
-        return final_hidden_states.view(num_tokens, hidden_size)
+        return final_hidden_states.view(batch_size, sequence_length, hidden_size)
```
Comment (kzawora-intel) [inline vllm/model_executor/models/mixtral.py:252]: this is HPU specific and will break on other accelerators. Please cover that with is_hpu.

File: vllm/model_executor/layers/fused_moe/fused_moe.py [hunk 3]
```diff
@@ -477,3 +480,31 @@ def fused_moe(
...
+
+    for expert_idx in range(num_experts):
+        padded_weight = padded_weights[expert_idx]
+        current_state_static = hidden_states.reshape(-1, D)
+        w_output = SiluAndMul()(torch.matmul(current_state_static, w1[expert_idx].transpose(0, 1)))
+        w_output = torch.matmul(w_output, w2[expert_idx].transpose(0, 1))
+        current_hidden_states_static = w_output * padded_weight
+        final_hidden_states += current_hidden_states_static
+
+    return final_hidden_states.view(-1, D)
```
Comment (kzawora-intel) [inline vllm/model_executor/layers/fused_moe/fused_moe.py:510]: nit: please add EOL at EOF

**Review by kzawora-intel — CHANGES_REQUESTED** [review]

File: vllm/model_executor/layers/fused_moe/fused_moe.py (hunk 3 above)
Comment (jkaniecki) [reply vllm/model_executor/layers/fused_moe/fused_moe.py]: Done

File: vllm/model_executor/models/mixtral.py (hunk 2 above)
Comment (jkaniecki) [reply vllm/model_executor/models/mixtral.py]: Done

File: vllm/model_executor/models/mixtral.py (hunk 1 above)
Comment (jkaniecki) [reply vllm/model_executor/models/mixtral.py]: Done

**Review by kzawora-intel — APPROVED** [review]
LGTM!

**Pull request merged**